src/hooks/useEthPrice.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+
+// Uses CoinGecko's free api to get eth price.
+// Caches the result in local storage for 5 minutes.
+export function useEthPrice() {
+  // Load the eth price from local storage into state
+  const [ethPrice, setEthPrice] = useState(localStorage.getItem('eth_price') || '0');
+
```

Comment:
I think I have a todo to look at useQuery for caching. But this seems like a case were we could use something like useQuery or something more sophisticated like Apollo (I've had pretty good luck with that on another project). To do the caching for us.